internal/podman/client.go

```diff
@@ -201,8 +217,54 @@ func (o *client) machineInit(ctx context.Context) error {
 		return nil
 	}
 
-	_, err = o.runPodman(ctx, "machine", "init")
-	return err
+	if _, err = o.runPodman(ctx, newMachineInitArgs()...); err != nil {
+		return err
+	}
+	return nil
+}
+
+func newMachineInitArgs() []string {
+	args := []string{"machine", "init"}
+	memory, _ := mem.VirtualMemory()
+	defaultMachineMemoryUint64, err := strconv.ParseUint(defaultMachineMemory, 10, 64)
+	if err != nil {
+		_, _ = log.Warning(err)
+		return args
+	}
+
+	if memory.Available > defaultMachineMemoryUint64 {
+		args = append(args, "--memory", defaultMachineMemory)
+	} else {
+		_, _ = log.Warningf(notEnoughMemoryAvailable, memory.Available, defaultMachineMemory)
+	}
+
+	cores, err := cpu.Counts(false)
+	if err != nil {
+		_, _ = log.Warning(err)
+		return args
+	}
+
+	defaultCPUs, err := strconv.Atoi(defaultMachineCPUs)
+	if err != nil {
+		_, _ = log.Warning(err)
+		return args
+	}
+
+	if cores >= defaultCPUs {
+		args = append(args, "--cpus", defaultMachineCPUs)
+	} else {
+		_, _ = log.Warningf(notEnoughCPUsAvailable, cores, defaultMachineCPUs)
+	}
+
+	return args
+}
+
+func (o *client) machineSet(ctx context.Context) error {
```

Comment:
do you need this?

Reply:
I think you're setting it on machine init

Reply:
ops you are right. Removing it